### PR TITLE
fix(workflow): remove unnecessary volume from coordinator

### PR DESCRIPTION
**Pull request by knight42**
**What this PR does / why we need it**:

coordinator no longer depends on Docker now.

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @zhujian7 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/workflow/workload/pod/builder.go [hunk 1]
```diff
@@ -236,16 +236,6 @@ func (m *Builder) CreateVolumes() error {
 		},
 	})
 
-	// Add host volume for host docker socket file for coordinator container.
-	m.pod.Spec.Volumes = append(m.pod.Spec.Volumes, corev1.Volume{
-		Name: common.HostDockerSockVolumeName,
```
Comment (zhujian7) [inline pkg/workflow/workload/pod/builder.go:241]: Can we delete `common.HostDockerSockVolumeName` or docker related const now?

Comment (knight42) [reply pkg/workflow/workload/pod/builder.go:241]: removed

**Pull request merged**